lib/ansible/modules/cloud/azure/azure_rm_mysqlfirewallrule_facts.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_mysqlfirewallrule_facts
+version_added: "2.8"
+short_description: Get Azure MySQL Firewall Rule facts.
+description:
+    - Get facts of Azure MySQL Firewall Rule.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+        required: True
+    server_name:
+        description:
+            - The name of the server.
+        required: True
+    name:
+        description:
+            - The name of the server firewall rule.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Get instance of MySQL Firewall Rule
+    azure_rm_mysqlfirewallrule_facts:
+      resource_group: resource_group_name
+      server_name: server_name
+      name: firewall_rule_name
+
+  - name: List instances of MySQL Firewall Rule
+    azure_rm_mysqlfirewallrule_facts:
+      resource_group: resource_group_name
+      server_name: server_name
+'''
+
+RETURN = '''
+rules:
+    description: A list of dictionaries containing facts for MySQL Firewall Rule.
```

Comment:
i dont know  firewall rule, just feel return info is so little, and seems not  very helpful. for firewall,  is there properties like allow/deny, port,  or application etc?